### PR TITLE
fix: Error doesn't catch correctly

### PR DESCRIPTION
**Pull request by gongpeione**
image

### CODE REVIEW

**Comment by vlio20** [comment]
Hi @gongpeione can you please create an issue for your PR? I would like to better understand the need for this fix.
Also, please add a unit test.

**Comment by gongpeione** [comment]
> Hi @gongpeione can you please create an issue for your PR? I would like to better understand the need for this fix. Also, please add a unit test.

Ok, I'll open an issue and add a unit test later, maybe this weekend.

**Comment by vlio20** [comment]
@gongpeione  any updates on this one?

File: src/timeout/timeout.spec.ts [hunk 1]
```diff
@@ -56,4 +56,24 @@ describe('timeout', () => {
     const t = new T();
     expect(await t.foo()).toEqual(1);
   });
+
+  it('should catch original exception after the method throw an error within provided ms', async () => {
```
Comment (vlio20) [inline src/timeout/timeout.spec.ts:60]: Can you also add a test that throws an error after the timeout and make sure that the timeout error is the one that is thrown?

**Review by vlio20 — CHANGES_REQUESTED** [review]

File: src/timeout/timeout.spec.ts (hunk 1 above)
Comment (gongpeione) [reply src/timeout/timeout.spec.ts:60]: Done

**Pull request merged**